File: DFS & BFS/pro_targetNumber.java
class Solution {
    static int cnt=0;
    public void dfs(int target, int sum,int[] numbers,int idx){
        if(idx == numbers.length){
           if(target == sum ) cnt++;
            return;
        }
        dfs(target,sum-numbers[idx],numbers,idx+1);
        dfs(target,sum+numbers[idx],numbers,idx+1);
    }
    public int solution(int[] numbers, int target) {
        int answer = 0;
        dfs(target,0,numbers,0);
        return cnt;
    }
}

File: DFS & BFS/pro_targetNumber.py
def solution(numbers, target):
    answer = 0
    def dfs(L,t):
        nonlocal answer
        if L==len(numbers):
            if t==target:
               answer+=1 
            return
        dfs(L+1,t+numbers[L]*(-1))
        dfs(L+1,t+numbers[L])
    dfs(0,0)    
    return answer